src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java

```diff
@@ -621,4 +625,49 @@ public static String composeBundleResourceRootFolter(BundleReader bundleReader)
             return composeSignedBundleFolder(bundleReader);
         }
     }
+
+    /**
+     * Returns the registry of an imageAddress or null if not present, or it's not an image address.
+     */
+    public static String extractImageAddressRegistry(String imageAddress) {
+        if (Strings.isNullOrEmpty(imageAddress)) {
+            return null;
+        }
+        try {
+            if (imageAddress.startsWith(DOCKER_IMAGE_TRANSPORT_PREFIX)) {
+                return DockerImage.fromString(imageAddress.substring(DOCKER_IMAGE_TRANSPORT_PREFIX.length())).getRegistry();
+            } else {
+                return DockerImage.fromString(imageAddress).getRegistry();
+            }
+        } catch (Exception ex) {
+            return null;
+        }
+    }
+
+    /**
+     * Determine the full qualified image address of an oci-based component.
+     */
+    public static String determineComponentFqImageAddress(
+            String componentImageAddress,
+            String bundleImageAddress,
+            String fallback) {
+        //~
+        String componentImageRegistry = extractImageAddressRegistry(componentImageAddress);
+        if (!Strings.isNullOrEmpty(componentImageRegistry)) {
+            return componentImageAddress;
+        }
+
```

Comment:
in general, here, we should insert log at debug level to understand why the method generated a certain FullyQualifiedImageAddress

Reply:
I'm not sure, this function is well covered by tests and this is a pretty normal case
We may add debug code on the callers  if necessary